app/models/katello/concerns/smart_proxy_extensions.rb

```diff
@@ -155,8 +155,7 @@ def pulp_mirror?
       end
 
       def pulp_master?
-        # use map instead of pluck in case the features aren't saved yet during create
-        self.features.map(&:name).include?(PULP_FEATURE)
+        self.smart_proxy_features.map(&:feature).any? { |f| PULP_FEATURE == f.name }
```

Comment:
Isn't this equivalent?

```suggestion
        self.has_feature?(PULP_FEATURE)
```

Reply:
Yes..This works well..Wasn't aware of this method. Will update! :+1:

Reply:
https://github.com/theforeman/foreman/commit/8f657a8422a1ab85defaa9a999cc79c32ccfb1c6 has been there for a while so surprised nobody noticed this given the author worked on katello :)